File: Cc.Core/Models/TabCity.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models
{
    public partial class TabCity
    {
        public long Id { get; set; }
        public DateTime InsertDate { get; set; }
        public string? InsertUser { get; set; }
        public DateTime UpdateDate { get; set; }
        public string? UpdateUser { get; set; }
        public long TabDistrictId { get; set; }
        public string Code { get; set; } = null!;
        public string? IstatCode { get; set; }
        public string? Description { get; set; }
        public string? CadastralCode { get; set; }

        public virtual TabDistrict TabDistrict { get; set; } = null!;
    }
}

File: Cc.Ui/src/models/tabCity.ts
import { Entity, resource, field, association } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';
import { TabDistrict } from './tabDistrict';
import { cache } from '../app/cache';
import { LookupKey } from '../core/lookup';

@resource('tabCity')
export class TabCity extends Entity {

    @field()
    id: number;
    @field()
    insertDate: string;
    @field()
    insertUser: string;
    @field()
    updateDate: string;
    @field()
    updateUser: string;
    @field()
    tabDistrictId: number;    
    @field()
    code: string;
    @field()
    istatCode: string;
    @field()
    description: string;
    @field()
    cadastralCode: string;
    
    @association(TabDistrict, 'tabDistrictId')
    tabDistrict: TabDistrict;

    guid: string;

    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): TabCity {

        let item = new TabCity();
        return item;
    }
}

@resource('tabCity')
export class TabCitySearchModel extends Entity implements Pagination {
    
    @field()
    districtId: number;
    @field()
    districtCode: string;
    @field()
    description: string;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };

    constructor(data?: any) {
        super()
        this.init(data);      
    }
}